src/SignalR/clients/ts/signalr/src/WebSocketTransport.ts

```diff
@@ -139,12 +138,7 @@ export class WebSocketTransport implements ITransport {
 
     public stop(): Promise<void> {
         if (this.webSocket) {
-            // Clear websocket handlers because we are considering the socket closed now
-            this.webSocket.onclose = () => {};
-            this.webSocket.onmessage = () => {};
-            this.webSocket.onerror = () => {};
-            this.webSocket.close();
-            this.webSocket = undefined;
+            this.cleanup();
```

Comment:
Nit: close() also calls cleanup(), and close() is called immediately after this. I'd just remove cleanup() and put the logic directly in close().